packages/common/src/bag3d/common/resources/database.py

```diff
@@ -1,35 +1,30 @@
-from dagster import resource, Field, Permissive
+from dagster import ConfigurableResource
 
 from pgutils import PostgresConnection, PostgresFunctions
 
-
 DatabaseConnection = PostgresConnection
 
 
-@resource(
-    config_schema={
-        "port": Field(int, description="Database port."),
-        "host": Field(
-            str, default_value="localhost", description="Database host to connect to."
-        ),
-        "user": Field(str, description="Database username."),
-        "password": Field(str, description="Database password."),
-        "dbname": Field(str, description="Database to connect to. It must exist."),
-        "other_params": Permissive(
-            description="Other connection parameters to be passed on to the database."
-        ),
-    },
-    description="Database connection.",
-)
-def db_connection(context):
-    conn = DatabaseConnection(
-        user=context.resource_config["user"],
-        password=context.resource_config["password"],
-        host=context.resource_config["host"],
-        port=context.resource_config["port"],
-        dbname=context.resource_config["dbname"],
-        **context.resource_config["other_params"],
-    )
-    # Create the utility Postgres functions
-    PostgresFunctions(conn)
-    return conn
+class DatabaseResource(ConfigurableResource):
+    """
+    Database connection.
+    """
+
+    host: str
+    user: str
+    password: str
+    dbname: str
+    port: str
+
```

Comment:
I removed **otherparams, I am not sure how to set it for ConfigurableResources.

Reply:
Yes we do need it, because we use `other_params` to pass connection parameters that we might not use, but others do, for example `sslmode`. So we need to have the possibility to forward any connection parameter that postgres (and psycopg) accepts.